OpenRA.Game/Game.cs

```diff
@@ -189,8 +192,20 @@ public static void RestartGame()
 			// Restart the game with the same replay/mission
 			if (replay != null)
 				JoinReplay(replayName);
-			else
+			else if (mission)
 				StartMission(uid, globalSettings.GameSpeedType, globalSettings.Difficulty);
+			else
+			{
+				JoinServer(IPAddress.Loopback.ToString(), CreateLocalServer(map), "");
+				LobbyInfoChanged += WidgetUtils.Once(() =>
```

Comment:
Can you combine this with `StartMission` (which was only introduced in the last PR, IIRC) to avoid duplication?